### DISCUSSION TITLE
How can i track argument changes in named route?

### DISCUSSION THREAD

**Discussion by mrnusa** [jonataslaw/getx#1103] [Q&A]
Hi, 

i have some item description pages accessible with the menu drawer. These are named routes the item id passed with 'Get.arguments'. 
My problem is because it is accessible from the menu the controller not disposed and when i tap with a different item i see the previous one. (also i have to set preventDuplicates: false to open it again)
How can i track the argument changes? The onInit, onReady not called because is an active controller.
Now i hotfix it with the following way:
In the view i use a GetBuilder, so i use didChangeDependencies callback to check Get.arguments changed or not, if yes i pass it to the controller to refresh data.
Any better way for that?
Or i have to forget named routes and i have to use 'Get.to(NextScreen());' and i have to pass the param to the view constructor?

Thank you!

**Comment by eduardoflorence** [comment]
In this case it would be better to pass the parameter through the constructor

**Reply by mrnusa** [reply]
Hi, thanks.

**Reply by mrnusa** [reply]
An another problem for the view constructor version:
In the view controller i cannot use the controller what i associated in the bindings. So i have to manually put it with Get.put? Or i forgot something?

**Reply by fer-ri** [reply]
Can you give some example code?

**Reply by mrnusa** [reply]
Now i init controller that way. This is the best use case?
`class TicketView extends GetView<TicketController> {`
  `TicketView({@required ticketId}) {`
    `Get.put(TicketController());`
   ` controller.getTicket(ticketId);`
 ` }`
`}`

And i call it from drawer that way:
`Get.to(() => TicketView(ticketId: _id,));`

thanks

**Reply by eduardoflorence** [reply]
Show your binding for this view

**Reply by mrnusa** [reply]
```
import 'package:get/get.dart';
import 'ticket_controller.dart';

class TicketBinding extends Bindings {
  @override
  void dependencies() {
    Get.lazyPut<TicketController>(
      () => TicketController(),
    );
  }
}

```

**Reply by fer-ri** [reply]
Why you have `Get.put(TicketController());` even it's already defined by your Binding?

**Reply by mrnusa** [reply]
Because at this point (in the view constructor) not yet initialized, if i remove it i receive TicketController not found error.

**Reply by fer-ri** [reply]
Are you sure you put your binding in the route?

**Reply by mrnusa** [reply]
yep:
 ```
abstract class Routes {
  static const TICKET = '/ticket';
}

class AppPages {
  static final routes = [
  GetPage(
        name: Routes.TICKET,
        page: () => TicketView(),
        binding: TicketBinding(),
      )
  ];
}
```
and it is working fine if i use named route

**Reply by fer-ri** [reply]
Yes, you should use a named route to make binding works.

By using `Get.to()` then actually you just initiate a new Widget Class and bypassing GetPage > binding.

**Reply by mrnusa** [reply]
Thanks, but the original question is still open. In this way (multiple description pages thats need param to work) what is the use case?
1, use named route and passes argument and refresh it with didChangeDependencies if needed.
or
2, use Get.to() to call the constructor and put the binding manually?
Or there is any better approach?

thanks:)

**Reply by mrnusa** [reply]
So, no any hope to work the bindings fine and handle attribute changes automatically?

**Comment by fbanyai** [comment]
Hi @mrnusa ,

I was suffering from a very similar scenario as yours - having the same page (view + controller) not being refreshed when called with different arguments.

After struggling with it for a few days I finally remembered of the property `tag` of the `GetBuilder` widget. And voilà! It worked like a charm!

Please notice that you must keep using the `preventDuplicates: false` in the `Get.to()` or `Get.off()` calls leading to the desired page.

```dart
class GroupPage extends StatelessWidget {
  final String groupGuid = Get.arguments['groupGuid'];

  @override
  Widget build(BuildContext context) {
    return GetBuilder<GroupController>(
        init: GroupController(groupGuid),
        tag: groupGuid,
        builder: (_) {
          return Scaffold(
            .
            .
            .
          )
        }
    );
  }
}
```
